src/net_processing.cpp

```diff
@@ -4354,7 +4354,9 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         const TxValidationState& state = result.m_state;
 
         if (result.m_result_type == MempoolAcceptResult::ResultType::VALID) {
-            ProcessValidTx(pfrom.GetId(), ptx, result.m_replaced_transactions.value());
+            Assume(result.m_replaced_transactions.has_value());
+            std::list<CTransactionRef> empty_replacement_list;
+            ProcessValidTx(pfrom.GetId(), ptx, result.m_replaced_transactions.value_or(empty_replacement_list));
```

Comment:
> it is unclear to me from the commit message and the code change how this change fits in the context.

(Note that this commit is a followup from #29619)

Yes, it should always have a value when the result is VALID. This is just adding a belt-and-suspenders juuust in case.

Reply:
Thanks! Great, I’ll attempt another more thorough review at a later time.

Reply:
Elaborated in commit message